java-frontend/src/main/java/org/sonar/java/resolve/Types.java

```diff
@@ -133,6 +135,22 @@ public int compare(Type type, Type t1) {
         return type.name().compareTo(t1.name());
       }
     });
+    Collections.sort(minimalCandidates, new Comparator<Type>() {
+      // Place classes before interfaces
+      @Override
+      public int compare(Type type, Type t1) {
```

Comment:
This should be merge with the other comparator to have only one pass on the sorting.